### PR TITLE
Feature 37 attach pictures to products

### PR DESCRIPTION
**Pull request by ignatIgnatov**

### CODE REVIEW

File: src/main/java/jewellery/inventory/mapper/ImageDataMapper.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.UUID;
+
+@Mapper(componentModel = "spring")
+public interface ImageDataMapper {
+  @Mapping(target = "productId", expression = "java(getProductId(image.getProduct()))")
+  ImageResponseDto toImageResponse(Image image);
+
+  default UUID getProductId(Product product) {
+    return product.getId();
```
Comment (berki0) [inline src/main/java/jewellery/inventory/mapper/ImageDataMapper.java:17]: this method is not used

File: src/main/resources/db/migration/V1_0__insert_user_admin.sql
Comment (berki0) [inline src/main/resources/db/migration/V1_0__insert_user_admin.sql:1]: change the migration name to V1.1

File: src/main/resources/db/migration/V2_0__update_resources.sql
Comment (berki0) [inline src/main/resources/db/migration/V2_0__update_resources.sql:1]: change the migration name to V1.2

File: src/main/resources/db/migration/V3_0__change_fields_in_product.sql
Comment (berki0) [inline src/main/resources/db/migration/V3_0__change_fields_in_product.sql:1]: change the migration name to V1.3

File: src/test/resources/static/img/Sample-jpg-image-10mb.jpg
Comment (berki0) [inline src/test/resources/static/img/Sample-jpg-image-10mb.jpg:1]: delete image

File: src/test/resources/static/img/pearl.jpg
Comment (berki0) [inline src/test/resources/static/img/pearl.jpg:1]: delete image

File: src/test/resources/static/img/test.txt
Comment (berki0) [inline src/test/resources/static/img/test.txt:1]: delete test.txt file

File: src/main/java/jewellery/inventory/service/ImageService.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
```
Comment (berki0) [inline src/main/java/jewellery/inventory/service/ImageService.java:24]: Optimize imports

Comment (berki0) [inline src/main/java/jewellery/inventory/service/ImageService.java]: Optimize imports

File: src/main/resources/db/migration/V1_0__insert_user_admin.sql
Comment (ignatIgnatov) [reply src/main/resources/db/migration/V1_0__insert_user_admin.sql:1]: Done!

File: src/main/resources/db/migration/V2_0__update_resources.sql
Comment (ignatIgnatov) [reply src/main/resources/db/migration/V2_0__update_resources.sql:1]: Done!

File: src/main/resources/db/migration/V3_0__change_fields_in_product.sql
Comment (ignatIgnatov) [reply src/main/resources/db/migration/V3_0__change_fields_in_product.sql:1]: Done!

File: src/main/java/jewellery/inventory/service/ImageService.java (hunk 2 above)
Comment (ignatIgnatov) [reply src/main/java/jewellery/inventory/service/ImageService.java]: Done!

File: src/test/resources/static/img/Sample-jpg-image-10mb.jpg
Comment (ignatIgnatov) [reply src/test/resources/static/img/Sample-jpg-image-10mb.jpg:1]: I use this file in the integration tests. If I delete them they won't work...

File: src/test/resources/static/img/pearl.jpg
Comment (ignatIgnatov) [reply src/test/resources/static/img/pearl.jpg:1]: Same thing

File: src/main/java/jewellery/inventory/service/ProductService.java [hunk 3]
```diff
@@ -50,17 +52,14 @@ public List<ProductResponseDto> getByOwner(UUID ownerId) {
...
-
-    return productMapper.mapToProductResponseDto(product);
-  }
-
   public Product getProduct(UUID id) {
     return productRepository.findById(id).orElseThrow(() -> new ProductNotFoundException(id));
   }
 
+  public ProductResponseDto getProductResponse(UUID id) {
+    return productMapper.mapToProductResponseDto(productRepository.findById(id).orElseThrow(() -> new ProductNotFoundException(id)));
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/ProductService.java]:
```suggestion
    return productMapper.mapToProductResponseDto(getProduct(id));
```

File: src/test/resources/application.properties [hunk 4]
```diff
@@ -12,4 +12,17 @@ cors.allowedHeaders=Authorization,Cache-Control,Content-Type
...
+# All files uploaded through the API will be stored in this directory
+image.folder.path=/tmp/Jms/Images/
+
+# Enable multipart uploads
+spring.servlet.multipart.enabled=true
+# Max file size
+spring.servlet.multipart.max-file-size=12MB
+# Max Request Size
+spring.servlet.multipart.max-request-size=12MB
+multipart.file.max.size=8
```
Comment (VladoKat) [inline src/test/resources/application.properties]: If these are the same as the ones in the src/main/resources/application.properties file, then we shouldn't need to specify them here as well

**Review by VladoKat — CHANGES_REQUESTED** [review]
Very minor changes requested. It looks good overall! 
I do have some consideration we should discuss maybe. They are around the filepath and name of the image:
1. Should we include the filePath of the image in the response to the user? Is it useful for them to know this?
2. Should we take the original name of the file and save it in our system or is it better to name the files ourselves. The reason behind this would be if someone uploaded a file with a name with some weird and/or malicious characters. Do we know if this can somehow break our system or mess up the folder where we save the files?

**Review by VladoKat — APPROVED** [review]
Looks good!

**Pull request merged**